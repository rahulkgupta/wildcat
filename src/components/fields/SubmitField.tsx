import SubmitField from '@src/util/form/fields/submit';

interface Props {
  field: SubmitField;
  submit: Function;
}

/**
 * Renders the react view for a {@link TextField}
 * uses the various functions of {@link TextField} to handle display and updates.
 * @param props contains the {@link TextField}
 */
const SubmitFieldView = (props: Props) => {
  const field = props.field;
  return (
    <input
      type="submit"
      value={field.getValue()}
      onClick={(event) => {
        props.submit();
      }}
      data-testid={field.getId()}
    ></input>
  );
};

export default SubmitFieldView;
