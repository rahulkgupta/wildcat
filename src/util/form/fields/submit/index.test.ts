import SubmitField from '.';

let submitField: SubmitField;

describe('SubmitField', () => {
  beforeEach(() => {
    submitField = new SubmitField(
      '1234',
      {
        value: '2343',
      },
      jest.fn(),
    );
  });

  test('getValue', () => {
    expect(submitField.getValue()).toMatchSnapshot();
  });

  test('toJSON', () => {
    expect(submitField.toJSON()).toMatchSnapshot();
  });
});
