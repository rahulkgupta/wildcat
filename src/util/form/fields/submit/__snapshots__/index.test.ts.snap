// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitField getValue 1`] = `"2343"`;

exports[`SubmitField toJSON 1`] = `
Object {
  "id": "1234",
  "value": "2343",
}
`;
